File: 3276-minimum-number-of-pushes-to-type-word-ii/3276-minimum-number-of-pushes-to-type-word-ii.cpp
class Solution {
public:
    int minimumPushes(string word) {
        int ans=0;
        vector<int> v(26,0);
        for(int i=0;i<word.size();i++) v[word[i]-'a']++;
        sort(v.begin(),v.end(),greater<int>());
        for(int i=0;i<26;i++) ans+=v[i]*((i+8)/8);
        return ans;
    }
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/3276-minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int n=word.length(),ans=0,push=8;
        int[] freq=new int[26];
        for(int i=0;i<n;i++) freq[word.charAt(i)-'a']++;
        Arrays.sort(freq);
        for(int i=25;i>=0;i--){
            ans+=freq[i]*(push/8);
            push++;
        }
        return ans;
    }
}